### PR TITLE
[aes, dv] V2S coverage improvements - Part 2

### LINKED ISSUE #13572: [aes] Forcing main/cipher core FSM signals sometimes doesn't trigger fatal alerts
@vogelpi 
if you look at PR #13571
and run /util/dvsim/dvsim.py hw/ip/aes/dv/aes_masking_sim_cfg.hjson -i aes_control_fi -r 1 -t xcelium  -s=1108401101

you will see tb.dut.u_aes_core.u_aes_control.gen_fsm[2].gen_fsm_n.u_aes_control_fsm_i.u_aes_control_fsm.cipher_in_ready_i
begin forced  high - and the two other interfaces continuing on as usual - not fatal is asserted.
this is one of the signals in the list you wanted me to check.
image

### PR DESCRIPTION
**Pull request by vogelpi**
This is a follow-up of #15907. The main changes are:
- Set constraints that are more useful for the FI tests, i.e., trigger more interesting state transitions and handshakes (e.g. with CTR mode FSM) and operations (decryption key generation) more frequently.
- Don't drive signal values that are less useful. More precisely, some handshake signals are backed up with a counter, driving these to 1 earlier doesn't have any effect. This resolves #13572.

The main result of this PR is that less FI tests timeout/fail which we need to reach the coverage goal and pass criteria for AES V2S.

### CODE REVIEW

**Review by weicaiyang — APPROVED** [review]
LGTM

**Comment by vogelpi** [comment]
Thanks for approving this @weicaiyang ! I've now rebased on master. Once CI passes and this can be merged if others don't mind.

**Review by andreaskurth — APPROVED** [review]
This is great work, thanks @vogelpi!

I've just run a full coverage run with this PR (and #15871 and #15907) applied and get results above the 90% threshold:

|  Score  |  Block  |  Branch  |  Statement  |  Expression  |  Toggle  |   Fsm   |  Assertion  |  CoverGroup  |
|:-------:|:-------:|:--------:|:-----------:|:------------:|:--------:|:-------:|:-----------:|:------------:|
| 96.15 % | 95.93 % | 90.80 %  |   98.37 %   |   92.92 %    | 96.33 %  | 97.90 % |   98.39 %   |   92.79 %    |

Pass rates have improved significantly and are now at least 93% (`aes_control_fi`).

**Review by ctopal — APPROVED** [review]

**Pull request merged**
